src/fastmcp/server/auth/proxy.py

```diff
@@ -0,0 +1,1059 @@
+
+        Delegates to the JWT verifier which handles signature validation,
+        expiration checking, and claims validation using the upstream JWKS.
+        """
+        result = await self._token_validator.verify_token(token)
+        if result:
+            logger.debug("Token validated successfully")
+        else:
+            logger.debug("Token validation failed")
+        return result
+
+    # -------------------------------------------------------------------------
+    # Token Revocation
+    # -------------------------------------------------------------------------
+
+    async def revoke_token(self, token: AccessToken | RefreshToken) -> None:
+        """Revoke token locally and with upstream server if supported.
+
+        Removes tokens from local storage and attempts to revoke them with
+        the upstream server if a revocation endpoint is configured.
+        """
+        # Clean up local token storage
+        if isinstance(token, AccessToken):
+            self._access_tokens.pop(token.token, None)
+            # Also remove associated refresh token
+            paired_refresh = self._access_to_refresh.pop(token.token, None)
+            if paired_refresh:
+                self._refresh_tokens.pop(paired_refresh, None)
+                self._refresh_to_access.pop(paired_refresh, None)
+        else:  # RefreshToken
+            self._refresh_tokens.pop(token.token, None)
+            # Also remove associated access token
+            paired_access = self._refresh_to_access.pop(token.token, None)
+            if paired_access:
+                self._access_tokens.pop(paired_access, None)
+                self._access_to_refresh.pop(paired_access, None)
+
+        # Attempt upstream revocation if endpoint is configured
+        if self._upstream_revocation_endpoint:
+            try:
+                async with httpx.AsyncClient(
+                    timeout=HTTP_TIMEOUT_SECONDS
+                ) as http_client:
+                    await http_client.post(
+                        self._upstream_revocation_endpoint,
+                        data={"token": token.token},
+                        auth=(
+                            self._upstream_client_id,
+                            self._upstream_client_secret.get_secret_value(),
+                        ),
+                    )
+                    logger.info("Successfully revoked token with upstream server")
+            except Exception as e:
+                logger.warning("Failed to revoke token with upstream server: %s", e)
+        else:
+            logger.debug("No upstream revocation endpoint configured")
+
+        logger.info("Token revoked successfully")
+
+    # -------------------------------------------------------------------------
+    # Custom Route Handling
+    # -------------------------------------------------------------------------
+
+    async def _handle_proxy_token_request(self, request: Request) -> JSONResponse:
```

Comment:
Is this actually being used and forwarding the verifier or is `_handle_idp_callback` being used instead?